Code_Exercises/Exercise_01_Compiling_with_SYCL/solution.cpp

```diff
@@ -11,11 +11,7 @@
 #define CATCH_CONFIG_MAIN
 #include <catch2/catch.hpp>
 
```

Comment:
So the ifdef here was added originally because for SYCL 2020 we should really be using `SYCL/sycl.hpp` as `CL/sycl.hpp` is deprecated, but DPC++ doesn't support that yet. It might be okay to switch back to `CL/sycl.hpp` for better compatibility, but I would worry it might encourage people to use the deprecated header. I'm tempted to say we keep the ifdef, and instead have something in the readme that talks about limitations for the different implementations. @ProGTX what do you think?

Reply:
Change made, using:
```
#if __has_include(<SYCL/sycl.hpp>)
#include <SYCL/sycl.hpp>
#else
#include <CL/sycl.hpp>
#endif
```
since the previous checks 
```
#if defined(SYCL_LANGUAGE_VERSION) && defined(__INTEL_LLVM_COMPILER)
```
Were less general and didn't work on all systems. 

```__has_include``` should work on all C++17 compilers (which is required for all SYCL implementations?).

Reply:
C++17 is required for SYCL 2020, so `__has_include` seems like the most elegant solution

Reply:
Should we update `Code_Exercises/Exercise_01_Compiling_with_SYCL/doc.md` as well?

Reply:
Yeah I like the `__has_include` solution 👍